src/main/java/com/example/demo/tournament/Tournament.java

```diff
@@ -0,0 +1,104 @@
+package com.example.demo.tournament;
+
+import com.example.demo.model.Match;
+import com.example.demo.model.Team;
+import com.example.demo.dao.MatchDAO;
+import com.example.demo.dao.TeamDAO;
+import lombok.AllArgsConstructor;
+import lombok.Value;
+import org.springframework.stereotype.Component;
+
+import java.rmi.UnexpectedException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+@Component
+@Value
+@AllArgsConstructor
+public class Tournament {
+
+    public boolean createAndSaveMatchesBasedOnListOfTeams(List<Team> teams) {
+        List<Match> matches = new ArrayList<>();
+
+        if (teams.size() % 2 == 0) {
+            Team firstTmpTeam;
+            Team secondTmpTeam;
+            int firstRandomValue;
+            int secondRandomValue;
+            int countMatchesAtTheRound = teams.size() / 2;
+            int cntOfMatches = teams.size() - 1;
+            int currentRound;
+            int cntOfRounds;
+            List<Integer> indexes = generateIndexList(teams.size());
+            List<Integer> tmpIndexes;
+            cntOfRounds = returnCntOfRounds(cntOfMatches);
+
+            for (int i = 1; i <= cntOfRounds; i++) {
+                currentRound = i;
+                tmpIndexes = new ArrayList<>();
+                for (int j = 0; j < countMatchesAtTheRound; j++) {
+                    firstRandomValue = getRandomValueFromArray(indexes);
+                    indexes.remove((Object) firstRandomValue);
+                    secondRandomValue = getRandomValueFromArray(indexes);
+                    indexes.remove((Object) secondRandomValue);
+
+                    firstTmpTeam = teams.get(firstRandomValue);
+                    secondTmpTeam = teams.get(secondRandomValue);
+
+                    Match match = new Match(firstTmpTeam.getName(), secondTmpTeam.getName(), currentRound, cntOfRounds, Match.generateScore());
+                    matches.add(match);
+
+                    if (getWinner(match).getName().equals(firstTmpTeam.getName())) {
+                        tmpIndexes.add(firstRandomValue);
+                    }
+                    else tmpIndexes.add(secondRandomValue);
+                }
+                indexes = new ArrayList<>(tmpIndexes);
+                countMatchesAtTheRound /= 2;
+            }
+        }
+        MatchDAO.save(matches);
+        return true;
+    }
+
+    private Integer getRandomValueFromArray(List<Integer> indexes) {
```

Comment:
https://stackoverflow.com/questions/8065532/how-to-randomly-pick-an-element-from-an-array

Why so many actions just to take random value?

Reply:
I agree, the code is not clear. I made a Set instead of List...